### ISSUE TITLE
Manually set content-type when using .attach

### ISSUE THREAD

**Issue by Glutch** [visionmedia/superagent#1152]
Hey!
I'm sending a request like this

```
request
    .post('http://website.com/upload')
    .attach('image', path)
    .send(payload)
    .set('content-type', 'multipart/form-data; boundary=----WebKitFormBoundaryKWf3FhTklvtPkprk')
```

However, the content-type gets overridden by superagents own value. Resulting in a res.req._headers like this

`content-type': 'multipart/form-data; boundary=--------------------------442535245526379898253153`

Is there any way i can force set the content-type manually?

Or perhaps figure out superagents content-type pre-post so i can use the same boundary in my .send(payload)

**Comment by pornel** [comment]
No, there's no way. Setting it will break things.

In the browser version the boundary is set automatically by `FormData` element and we have no control over it.

**Comment by Glutch** [comment]
@pornel So theres literally no way i can use the same boundary in my payload as my header?
If no, can i use some other framework to accomplish this?

**Comment by pornel** [comment]
You can't use `attach()` and `send()` at the same time. You can either use your own content-type and your own payload with `.send().set()` OR you can let all of it being 100% auto-generated by the browser by using `.attach().field()`. They don't mix.

**Comment by Glutch** [comment]
@pornel Okey, so theres still a chance! Can you send images with .send()? :) (if so how?)

**Comment by pornel** [comment]
The hard way. Generate boundary, content-type header, and append image's binary data. You'll need to get Uint8Array from image Blob (using FileReader).

I'd rather do it the other way — why do you use your raw, custom payload? Can you send it as files or form fields instead?

**Comment by Glutch** [comment]
@pornel I'm trying to upload a picture to a website (not mine), i've checked the network requests and i need to provide the image and a payload. 

The payload consists of information regarding the image, like this

```
------WebKitFormBoundaryKWf3FhTClwfaDpre
Content-Disposition: form-data; name="image"; filename="2684337397.jpg"
Content-Type: image/jpeg
```

And the content-type (WebKitFormBoundaryKWf3FhTClwfaDpre) must match. I'm not sure how to do it

**Comment by pornel** [comment]
What do you use .send() for?

Usually you only need

```js
request
    .post('http://website.com/upload')
    .attach("image", image, "2684337397.jpg")
```

and nothing else. The boundary is supposed to be random, so needing it to match anything is very unusual.

**Comment by Glutch** [comment]
@pornel The payload also consists of eventvalidation and some viewstate string, so it seems to be required

**Comment by pornel** [comment]
You can add other data using `.field()` - it does the same thing as `.send()`, but uses form-data instead of JSON that can't send files.

**Comment by Glutch** [comment]
@pornel I just got it working, thank you a lot!

I've got a new, simpler problem. i'm sending some text via .send(), the site uses charset ISO-8859-1. I tried setting the charset with

```
const charset = require('superagent-charset')
charset(request)
.charset('ISO-8859-1')
```

This has worked previously when **reading** the response. But when i'm sending, the text coming up on the site is encoded weirdly. å showing up as Ã¥ etc. Is there a specific method of setting the charset of .send()?

_Should i make a new issue for this?_

Thanks a lot
